### DISCUSSION TITLE
Reading a variable created in OJS back into Python?

### DISCUSSION THREAD

**Discussion by paniterka** [quarto-dev/quarto-cli#3792] [Q&A] [answered]
Hello, 

I would like to run a Quarto notebook where I create a beeswarm chart in the following way: 
1. read in and manipulate the data using Python 
2. read manipulated data in OJS and use one of the D3 functions to calculate the beeswarm coordinates 
3. give those coordinates back to Python and create the visualization using Matplotlib or Plotly 

I am currently trying to leverage the algorithm from [this OJS notebook](https://observablehq.com/@jtrim-ons/non-force-directed-beeswarm-of-mobility-version-3-no-union-o) to perform step 2. 

While I can use `ojs_define` to achieve steps 1 and 2, I haven't found a way to go to step 3 (transfer the variables from the OJS engine back into Python).  I have tried to use 
```
%%javascript
IPython.notebook.kernel.execute("foo=56");
```
and similar, as described [here](https://stackoverflow.com/questions/37172978/how-to-pass-variables-from-javascript-to-python-in-jupyter), but it never worked from within a `.qmd` file, neither was I able to make the coupling between Python and OJS work in a `.ipynb` file. 

All the examples I was able to find only show how to pass the data from R/Python into JS, and not in reverse. Is there a way to do it in Quarto? Being able to access the D3 data manipulation function would be a very useful feature for creating complex charts without having to learn a JS-based visualization library.

**Comment by rgaiacs** [comment, marked as answer]
There is no way to pass data from OJS to Python or R. OJS is a client side library. This means that all OJS code runs in the web browser of the user reading your document instead of the machine the is converting the Quarto document to HTML.

You can try to use https://github.com/mgymrek/pybeeswarm.

**Reply by paniterka** [reply]
Thank you, I'll look for another way to achieve this :)
